pkgs/development/python-modules/seqdiag/fix_test_generate.patch

```diff
@@ -0,0 +1,25 @@
+diff --git a/src/seqdiag/tests/test_generate_diagram.py b/src/seqdiag/tests/test_generate_diagram.py
+index cdd340f..a1dd2d4 100644
+--- a/src/seqdiag/tests/test_generate_diagram.py
++++ b/src/seqdiag/tests/test_generate_diagram.py
+@@ -13,19 +13,5 @@
+ #  See the License for the specific language governing permissions and
+ #  limitations under the License.
+ 
+-import os
+-
+ from blockdiag.tests.test_generate_diagram import (get_diagram_files,
+-                                                   testcase_generator)
+-
+-import seqdiag.command
+-
+-
+-def test_generate():
+-    mainfunc = seqdiag.command.main
+-    basepath = os.path.dirname(__file__)
+-    files = get_diagram_files(basepath)
+-    options = []
+-
+-    for testcase in testcase_generator(basepath, mainfunc, files, options):
+-        yield testcase
++                                                   test_generate_with_separate)
```

Comment:
Could you help me understand what this patch is doing? What’s the difference in behaviour here?

Reply:
`testcase_generator` was replaced with `test_generate_with_separate` with `blackdiag` migrate-to-pytest patch
Since it uses a pytest decoration (`@pytest.mark.parametrize`) it is collected automatically by pytest. All the logic is done through the single `import` statement